Improve `spice dataset` documentation (#734)
spiceai/docs: spiceaidocs/docs/cli/reference/dataset.md

---
 title: 'dataset'
 sidebar_label: 'dataset'
 pagination_prev: null
 pagination_next: null
 ---
Configure a Spice dataset.

### Usage
```shell
spice dataset [command]
```

 Available `command`s:

>  - `configure`: Create/configure a dataset directly from the command-line, including customizing components such as Data Connector (`from` in a Spicepod) and acceleration along with other metadata.

Reviewer: ```suggestion
 - `configure`: Create/configure a dataset directly from the command-line, including customizing components such as whether to add acceleration to the connector.
```

 #### Flags

 - `-h`, `--help` Print this help message

 ### Example

 When running `spice dataset configure`, Spice will prompt for four inputs:
 1. The name of the dataset, labelled by `(1)` below.
 2. The description of the dataset, labelled by `(2)` below.
 3. The source of the dataset, labelled by `(3)` below. Consult [Spice's supported data connectors](/components/data-connectors) to see possible values for this field. 
 4. Whether or not to enable acceleration for this dataset, labelled by `(4)`. The default value for this input is `y`, enabling acceleration for this dataset. Learn more about acceleration in the [dataset acceleration reference](/components/data-accelerators).

 ```shell
> spice dataset configure

 2024/12/18 01:06:32 INFO dataset name: sample-project
 taxi-trips # (1)
 2024/12/18 01:06:59 WARN Dataset names with hyphens should be quoted in queries:
 i.e. SELECT * FROM "remote-source"
 description: Taxi trips in s3 # (2)
 from: s3://spiceai-demo-datasets/taxi_trips/2024/  # (3)
 2024/12/18 01:075 INFO locally accelerate (y/n)? (y)
 n # (4)
 2024/12/18 01:07:32 INFO Saved datasets/remote-source/dataset.yaml

>  ```

Reviewer 2: It might be easier to read to separate the datset configure output to the log output, for example, something like.
```
spice dataset configure
dataset name: (spiceai) test
description: test dataset
from: s3://spiceai-demo-datasets/taxi_trips/2024/
file_format (parquet/csv) (parquet) parquet
locally accelerate (y/n)? (y) y
2025/01/10 14:07:46 INFO Saved datasets/test/dataset.yaml
```

Author: I included the logs because I wanted to replicate exactly what the user would see when running the command. I didn't want excluding the logs to result in any sources of confusion for the user. 

Let me know what you think, and I'll adjust accordingly.

Reviewer 2: Yeah we 'd like to include the exact logs that would be seen by user. However my log looks like the one in my previous comment, and my spice is built from latest trunk. Could you help provide the version of spice that you ran to get the log using `spice version`? Thanks

Author: Oh, I see. I think when I wrote this change earlier I was on release candidate 2, not 4. I've upgraded now and I'm getting the same output as you. I'll go update my changes in my PR.
